api/src/controllers/post/inspection.js

```diff
@@ -0,0 +1,178 @@
+    inpsection.write = ['sysadmin'];
+
+    let savedInspection = null;
+    try {
+        savedInspection = await inpsection.save();
+    } catch (e) {
+        return {
+            status: 'failure',
+            object: inpsection,
+            errorMessage: e
+        }
+    }
+
+    var observables = [];
+    incomingObj.InspectionLNG && observables.push(this.createLNG(args, res, next, incomingObj.InspectionLNG, savedInspection._id));
+    incomingObj.InspectionNRCED && observables.push(this.createNRCED(args, res, next, incomingObj.InspectionLNG, savedInspection._id));
+
+    var flavourRes = null;
+    try {
+        observables.length > 0 && (flavourRes = await Promise.all(observables));
+    } catch (e) {
+        flavourRes = {
+            status: 'failure',
+            object: observables,
+            errorMessage: e
+        }
+    }
+
+    return {
+        status: 'success',
+        object: savedInspection,
+        flavours: flavourRes
+    }
+};
+
+// Example of incomingObj
+/**
+ *  {
+ *      _master: '5e1e7fcd20e4167bcfc3daa7'
+ *      description: 'lng description',
+ *      ...
+ *      addRole: 'public'
+ *  }
+ */
+exports.createLNG = async function (args, res, next, incomingObj, masterId) {
+    var InspectionLNG = mongoose.model('InspectionLNG');
+    var inpsectionLNG = new InspectionLNG();
+
+    inpsectionLNG._schemaName = 'InspectionLNG';
+
+    // We must have a valid master ObjectID to continue.
+    masterId && ObjectId.isValid(masterId) && (inpsectionLNG._master = new ObjectId(masterId));
+    if (!inpsectionLNG._master) {
+        return {
+            status: 'failure',
+            object: incomingObj,
+            errorMessage: 'incomingObj._master was not valid ObjectId'
+        }
+    }
+
+    inpsectionLNG.read = ['sysadmin'];
+    inpsectionLNG.write = ['sysadmin'];
+    // If incoming object has addRole: 'public' then read will look like ['sysadmin', 'public']
+    incomingObj.addRole && incomingObj.addRole === 'public' && inpsectionLNG.read.push('public') && (inpsectionLNG.datePublished = new Date());
+
+    incomingObj.description && (inpsectionLNG.description = incomingObj.description);
+
+    inpsectionLNG.dateAdded = new Date();
+
+    try {
+        var savedInspectionLNG = await inpsectionLNG.save();
+        return {
+            status: 'success',
+            object: savedInspectionLNG
+        }
+    } catch (e) {
+        return {
+            status: 'failure',
+            object: inpsectionLNG,
+            errorMessage: e
+        }
+    }
+};
+
+// Example of incomingObj
+/**
+ *  {
+ *      _master: '5e1e7fcd20e4167bcfc3daa7'
+ *      description: 'nrced description',
+ *      ...
+ *      addRole: 'public'
+ *  }
+ */
+exports.createNRCED = async function (args, res, next, incomingObj, masterId) {
+    var InspectionNRCED = mongoose.model('InspectionNRCED');
+    var inpsectionNRCED = new InspectionNRCED();
+
+    inpsectionNRCED._schemaName = 'InspectionNRCED';
```

Comment:
Not required, but its not a bad idea to save strings like 'InspectionNRCED', and similar, as constants at the top level.  That way its easy to update every instance of its use at once, without having to track them all down.  Could use it in the mongoose.model() call as well, etc.

If these schema names are used across multiple files, then they could also be moved into a separate enum or constants file, so theres only 1 place that needs to be updated across the entire api.